libvast/builtins/endpoints/query.cpp

```diff
@@ -181,13 +150,68 @@ struct query_manager_state {
   std::optional<pipeline_executor> pipeline_executor_;
 
   std::string create_response() {
-    auto [drained_events, events_as_json]
-      = drain_buffer(processed_slices, limit);
-    shippable_events_count -= drained_events;
-    shipped_events_count += drained_events;
-    auto events = detail::split(events_as_json, "\n");
-    return fmt::format("{{\"position\": {}, \"events\": [\n{}]}}\n",
-                       shipped_events_count, fmt::join(events, ",\n"));
+    auto printer = json_printer{{.oneline = true}};
+    auto result = std::string{};
+    auto out_iter = std::back_inserter(result);
+    auto seen_schemas = std::unordered_set<type>{};
+    auto written = size_t{0};
+    out_iter = fmt::format_to(out_iter, "{{\"events\":[");
```

Comment:
I think we can just initialize result with "{"events":[" instead of initializing result + calling fmt.
Not a big deal, one line less to read though :D

Reply:
I made this change.